### ISSUE TITLE
`invalid-type-arguments` - false negative when type takes no type arguments

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3635]
### Summary

`main.py`:

```py
foo: str[int] # no error
```

[Playground](https://play.ty.dev/17b1937b-0b93-4354-8805-bce5518d95eb)

### Version

1a34347c2
